src/test/spec/v2_runner/mod.rs

```diff
@@ -271,7 +271,8 @@ pub async fn run_v2_test(test_file: TestFile) {
                                 .unwrap_or_else(|| panic!("ClientSession is not pinned"));
 
                             fail_point_guards.push(
-                                internal_client
+                                client
+                                    .deref()
```

Comment:
As far as I can tell it doesn't strictly matter here whether the failpoint is enabled on `internal_client` or the `EventClient`, but for consistency with the other failpoint we enable I changed this to the `EventClient`.

Reply:
Hm I agree we should be consistent with which client we're using for failpoints but I think it would make more sense to do all of them with the `internal_client` rather than the `EventClient`. In general we just use the `EventClient` for operations that need to be tracked for command monitoring and the internal client is used for test setup and test runner operations.